### PR TITLE
Re-Add (Local) support for nerd fonts

### LINKED ISSUE #15: Support nerd fonts
It would be nice, if [nerd fonts](https://github.com/ryanoasis/nerd-fonts) could be supported. These are quite common, if people play with their setup.

image

I haven't looked into it, but maybe it's enough to include one of the fonts and configure xterm to use it.

### PR DESCRIPTION
**Pull request by matthewstrasiotto**
Resolves #15 .

The changes made here are (mostly) copied across from https://github.com/elisescu/tty-server/pull/16 , with slight modifications to fit new path / routing structures.

As yet - only tested on local session due to #24 , and I'm not sure about the cachability of the new (heavy) assets.

I'll run a local proxy & see if I need to replicate the static asset structure between the two.

For now, they appear at `localhost:8000/s/local/static/fonts/`. I imagine that in public sessions they will appear at `url/s/[sessionid]/static/fonts/`.

Ideally these would sit behind some cloudflare/cloudfront cdn type service, as they're not lightweight dependencies.

### CODE REVIEW

**Comment by matthewstrasiotto** [comment]
Tested via local proxy- It seems to work perfectly fine behind the proxy, though I can't comment on the performance / whether different sessions need to backhaul these assets for each session between the sharing session to the proxy or not.

**Comment by matthewstrasiotto** [comment]
Latest pushes change the font source to a CDN for (ideally) reduced load on the server, and no extra load on the proxy as a result of delivering these font assets. 

There's a small loading time for the real fontset to arrive locally, wherein it falls back to courier-new, then system monospace. 

For now, this isn't ready to merge until I can squash all inclusions of the font assets since they're large, and deprecated, & shouldn't make it into the main repo index.

**Comment by matthewstrasiotto** [comment]
I've migrated the font assets to a CDN with an external stylesheet that references them corectly.
This alleviates annoying config between the go server's route config, and webpacks's asset packing, keeps the repo from bloating with binary assets, & should impose no extra load on the server.

**Review by elisescu — CHANGES_REQUESTED** [review]
I had a few inline comments - there are some leftovers that can be removed, as they don't seem to be needed with this new CDN change.
I like the CDN change better than including the fonts in the binary. 
I didn't forget about your original change in tty-server :) , and wanted to include it in this version 2 but it became very complex to mess with the paths of resources given how they are routed by the proxy. Also the binary became too big and I decided to leave them out - I want to keep the binary as small as possible.
But great idea with having them on a CDN, and a fallback font to use if/when CDN is down.

**Comment by matthewstrasiotto** [comment]
> I didn't forget about your original change in tty-server :) , and wanted to include it in this version 2 but it became very complex to mess with the paths of resources given how they are routed by the proxy.

Trust me- *I know*

Though the CDN is overall probably going to be better (less load on proxies & less overhead bandwidth from session hosts), the main motivation for the switch was to avoid stuffing with the routing paths.

File: server/frontend/webpack.config.js [hunk 1]
```diff
@@ -26,7 +26,21 @@ let mainConfig  = {
...
+                loader: 'file-loader',
+                options: {
+                  name: '[name].[ext]',
+                  outputPath: 'fonts/',
+                  esModule: false,
+                  publicPath: './static/fonts/'
+                }
+              }
+            ]
+          }
```
[the remark this answers is not in the document]
Comment (matthewstrasiotto) [reply server/frontend/webpack.config.js]: Done

**Comment by matthewstrasiotto** [comment]
@elisescu , thank you for the observations, you're absolutely right, the dependencies added to `package.json`, `package-lock.json` & `webpack.config.js` were redundant.

**Review by elisescu — APPROVED** [review]
Looks good :)

**Comment by elisescu** [comment]
Thank you for your contribution

**Pull request merged**